tests/test_group.rs

```diff
@@ -2,233 +2,242 @@ use openmls::prelude::*;
+        match group_alice.apply_commit(mls_plaintext_commit, epoch_proposals, &[]) {
+            Ok(_) => {}
+            Err(e) => panic!("Error applying commit: {:?}", e),
+        };
+        let ratchet_tree = group_alice.tree().public_key_tree_copy();
+
+        // Bob creates group from Welcome
+        let _group_bob = match MlsGroup::new_from_welcome(
+            welcome_bundle_alice_bob_option.unwrap(),
+            Some(ratchet_tree),
+            bob_key_package_bundle,
         ) {
-        Ok(c) => c,
-        Err(e) => panic!("Error creating commit: {:?}", e),
-    };
-    let commit = match &mls_plaintext_commit.content {
-        MLSPlaintextContentType::Commit((commit, _)) => commit,
-        _ => panic!(),
-    };
-    assert!(commit.path.is_none() && kpb_option.is_none());
-
-    // Alice applies the Commit without the forced self-update
-    match group_alice_1234.apply_commit(mls_plaintext_commit, epoch_proposals, &[]) {
-        Ok(_) => {}
-        Err(e) => panic!("Error applying commit: {:?}", e),
-    };
-    let ratchet_tree = group_alice_1234.tree().public_key_tree_copy();
-
-    // Bob creates group from Welcome
-    let _group_bob_1234 = match MlsGroup::new_from_welcome(
-        welcome_bundle_alice_bob_option.unwrap(),
-        Some(ratchet_tree),
-        bob_key_package_bundle,
-    ) {
-        Ok(group) => group,
-        Err(e) => panic!("Error creating group from Welcome: {:?}", e),
-    };
-
-    assert_eq!(
-        group_alice_1234.tree().public_key_tree(),
-        group_alice_1234.tree().public_key_tree()
-    );
-
-    // Alice updates
-    let alice_update_proposal = group_alice_1234.create_update_proposal(
-        group_aad,
-        &alice_credential_bundle,
-        alice_update_key_package.clone(),
-    );
-    let proposals = vec![alice_update_proposal];
-
-    // Only UpdateProposal
-    let (commit_mls_plaintext, _welcome_option, kpb_option) = match group_alice_1234.create_commit(
-        group_aad,
-        &alice_credential_bundle,
-        proposals.clone(),
-        false, /* force self update */
-    ) {
-        Ok(c) => c,
-        Err(e) => panic!("Error creating commit: {:?}", e),
-    };
-    let (commit, _confirmation_tag) = match &commit_mls_plaintext.content {
-        MLSPlaintextContentType::Commit((commit, confirmation_tag)) => (commit, confirmation_tag),
-        _ => panic!(),
-    };
-    assert!(commit.path.is_some() && kpb_option.is_some());
-
-    // Apply UpdateProposal
-    group_alice_1234
-        .apply_commit(
-            commit_mls_plaintext.clone(),
-            proposals,
-            &[kpb_option.unwrap()],
-        )
-        .expect("Error applying commit");
+            Ok(group) => group,
+            Err(e) => panic!("Error creating group from Welcome: {:?}", e),
+        };
+
+        assert_eq!(
+            group_alice.tree().public_key_tree(),
+            group_alice.tree().public_key_tree()
+        );
```

Comment:
I'm confused. This seem trivial to be trivially true for me.

It should probably be Bob's tree? Maybe we should have Bob also perform the commit? That should work as well.

Reply:
Yeah, that's a regression from all the merging...